kong/core/globalpatches.lua

```diff
@@ -49,6 +49,27 @@ return function(options)
 
 
 
+  do -- implement `sleep` in the `init_worker` context
+
+    -- initialization code regularly uses the shm and locks.
+    -- the resty-lock is based on sleeping while waiting, but that api
+    -- is unavailable. Hence we implement a BLOCKING sleep, only in
+    -- the init_worker context.
+    local get_phase= ngx.get_phase
+    local ngx_sleep = ngx.sleep
+    local alternative_sleep = require("socket").sleep
+    ngx.sleep = function(s)
+      if get_phase() == "init_worker" then
+        ngx.log(ngx.WARN, "executing a blocking 'sleep' (", s, " seconds)")
+        return alternative_sleep(s)
+      end
+      return ngx_sleep(s)
+    end
+
+  end
```

Comment:
Maybe we should forbid the use of lua-cassandra in the init_worker context. Since it can be used in the init context instead, and since init_worker does not allow us to use a mutex, it seems reasonable to do so.